ci/steps/attach_logs.py

```diff
@@ -1,179 +1,243 @@
+                            version=version,
+                            gardenlinux_epoch=int(gardenlinux_epoch),
+                            architecture=glci.model.Architecture(architecture),
+                            platform=platform,
+                            modifiers=modifiers,
+                        )
+
+                    manifest = glci.util.find_release(
+                        release_identifier=release_identifier,
+                        s3_client=s3_client,
+                        bucket_name=s3_bucket_name,
+                    )
+                    if manifest:
+                        print(f'Found manifest {manifest.s3_key} with logs {manifest.logs}')
+                        if manifest.logs:
+                            log_files.add(manifest.logs)
+                    else:
+                        print(f'Manifest for platform {platform}, \
+                            {release_identifier.canonical_release_manifest_key()} not found.')
 
-        # upload log and attach to manifest
-        # copy manifest and attach test_results
-        s3_key = upload_file_gen.s3_key
-        print(f'uploaded zipped logs to {s3_key=}')
-        new_manifest = manifest.with_logfile(s3_key)
-
-        # upload manifest
-        manifest_path_suffix = manifest.canonical_release_manifest_key_suffix()
-        manifest_path = f'{glci.model.ReleaseManifest.manifest_key_prefix}/{manifest_path_suffix}'
-        glci.util.upload_release_manifest(
-          s3_client=s3_client,
-          bucket_name=s3_bucket_name,
-          key=manifest_path,
-          manifest=new_manifest,
+        print(f'Attaching {len(log_files)} logs to manifest')
+        new_manifest_set = manifest_set.with_logfiles(tuple(log_files))
+
+        upload_release_manifest_set = glci.util.preconfigured(
+            func=glci.util.upload_release_manifest_set,
+            cicd_cfg=cicd_cfg,
         )
-    else:
-        upload_file_gen = _upload_package_file(
-            log_file_path=log_path,
-            s3_client=s3_client,
-            s3_bucket_name=s3_package_bucket_name,
-            s3_key='last_package_build_log.zip',
+        print(f'Uploading manifest set to {manifest_set_key}')
+        upload_release_manifest_set(
+            key=manifest_set_key,
+            manifest_set=new_manifest_set,
         )
-        s3_key = upload_file_gen.s3_key
-        print(f'uploaded zipped logs to {s3_key=}')
 
-    # write a file with the download URL so that it can be later added to the email
-    with open(os.path.join(repo_dir, 'log_url.txt'), 'w') as f:
-        f.write(f'https://gardenlinux.s3.eu-central-1.amazonaws.com/{s3_key}')
+    # if we are in a Release build logs mut be always attached and a missing manifest is a
+    # hard error
+    elif glci.model.BuildType.RELEASE in build_target_set:
+        print('Could not find release-manifest-set in release build, exit with failure')
+        return False
+
+    return True
 
 
+# Logging Concept:
```

Comment:
rather use a multiline-str? this way, it is discovered as a doc-str (either on module-level, or as first segment of the `upload_logs` function)

Reply:
Done!